src/eos-updater-poll-lan.c

```diff
@@ -0,0 +1,920 @@
+                                    records,
+                                    valid,
+                                    error))
+    return FALSE;
+  if (!*valid)
+    return TRUE;
+  *valid = txt_v1_handler_checks (lan_data,
+                                  swbf,
+                                  records);
+  return TRUE;
+}
+
+static gboolean
+txt_v2_handler_risky_checks (LanData *lan_data,
+                             GHashTable *records,
+                             gboolean *valid,
+                             GError **error)
+{
+  if (!check_ostree_path (lan_data,
+                          g_hash_table_lookup (records, eos_avahi_v2_ostree_path ()),
+                          valid,
+                          error))
+    return FALSE;
+  if (!*valid)
+    // TODO: message
+    return TRUE;
+  return TRUE;
+}
+
+static gboolean
+txt_v2_handler_checks (LanData *lan_data,
+                       EosServiceWithBranchFile *swbf,
+                       GHashTable *records)
+{
+  const gchar *dl_time;
+  g_autoptr(GDateTime) txt_utc = NULL;
+
+  dl_time = g_hash_table_lookup (records, eos_avahi_v2_branch_file_timestamp ());
+  switch (check_dl_time (lan_data, dl_time, &txt_utc))
+    {
+    case DL_TIME_OLD:
+      eos_service_with_branch_file_get_from_lan_data (swbf, lan_data);
+      return TRUE;
+
+    case DL_TIME_NEW:
+      swbf->declared_download_time = g_steal_pointer (&txt_utc);
+      return TRUE;
+
+    case DL_TIME_INVALID:
+      // TODO: message
+      return FALSE;
+
+    case DL_TIME_OUT_OF_RANGE:
+      // TODO: message
+      return FALSE;
+    }
+
+  // TODO: message
+  g_assert_not_reached ();
+  return FALSE;
+}
+
+static gboolean
+txt_v2_handler (LanData *lan_data,
+                EosServiceWithBranchFile *swbf,
+                gboolean *valid,
+                GError **error)
+{
+  g_autoptr(GHashTable) records = NULL;
+  TxtRecordError txt_error = get_unique_txt_records (swbf->service->txt,
+                                                     &records,
+                                                     eos_avahi_v2_ostree_path (),
+                                                     eos_avahi_v2_branch_file_timestamp (),
+                                                     NULL);
+  if (txt_error != TXT_RECORD_OK)
+    {
+      // TODO: message
+      *valid = FALSE;
+      return TRUE;
+    }
+  if (!txt_v2_handler_risky_checks (lan_data,
+                                    records,
+                                    valid,
+                                    error))
+    return FALSE;
+  *valid = txt_v2_handler_checks (lan_data,
+                                  swbf,
+                                  records);
+  return TRUE;
+}
+
+static ServiceCheck
+get_txt_handler (const gchar *txt_version)
+{
+  guint64 v;
+  const gchar *end = NULL;
+
+  errno = 0;
+  v = g_ascii_strtoull (txt_version, (gchar **)&end, 10);
+
+  if (errno != 0 || end == NULL)
+    return NULL;
+
```

Comment:
If the string was something like "123xyz", then it will return 123 and end will point to "x".

I'm not sure if `g_ascii_strtoull` ever returns `NULL` for the `end` variable. I'm just paranoid here.